### PR TITLE
Judgment  import resource update

### PR DESCRIPTION
**Pull request by actlikewill**
- Standardize the judgments bulk import resource esp. as regards to case numbers.

### CODE REVIEW

File: peachjam/resources.py [hunk 1]
```diff
@@ -129,12 +132,26 @@ def before_import_row(self, row, **kwargs):
         DocumentNature.objects.get_or_create(name=row["nature"])
 
 
+class JudgesWidget(ManyToManyWidget):
+    def clean(self, value, row=None, *args, **kwargs):
+        judges = list(map(" ".join, list(map(str.split, value.split(self.separator)))))
```
Comment (longhotsummer) [inline peachjam/resources.py:137]: I'm not clear what this is doing. Can you add a comment? Also, list comprehensions are generally easier to read and more pythonic than using `map`.

File: peachjam/resources.py [hunk 2]
```diff
@@ -146,31 +163,51 @@ def before_import_row(self, row, **kwargs):
                 defaults={"name": row["court_obj"]["name"]},
             )
 
+        source_files = row["source_url"].split("|")
+        docx = re.compile(r".docx?$")
```
Comment (longhotsummer) [inline peachjam/resources.py:167]: escape `.` because it's special

File: peachjam/resources.py [hunk 3]
```diff
@@ -146,31 +163,51 @@ def before_import_row(self, row, **kwargs):
...
+        pdf = re.compile(r".pdf$")
+        for file in source_files:
+            match_docx = docx.search(file)
+            match_pdf = pdf.search(file)
+            if match_docx:
+                row["source_url"] = file
+            elif match_pdf:
+                row["source_url"] = file
+
```
Comment (longhotsummer) [inline peachjam/resources.py:176]: this doesn't make any sense -- you're setting `source_url` to the same thing each time.

File: peachjam/resources.py [hunk 4]
```diff
@@ -146,31 +163,51 @@ def before_import_row(self, row, **kwargs):
...
-                    matter_type, _ = MatterType.objects.get_or_create(
-                        name=row["matter_type"]
-                    )
-                CaseNumber.objects.create(
-                    document=judgment,
-                    number=case_number[0],
-                    year=case_number[1],
-                    matter_type=MatterType.objects.get(name=row["matter_type"]),
+        elif row["case_numbers"]:
+            for c in list(map(str.strip, row["case_numbers"].split("|"))):
```
Comment (longhotsummer) [inline peachjam/resources.py:187]: Can you include a comment showing the type of input you're expecting?

**Review by longhotsummer — COMMENTED** [review]
I think it'd be useful to start adding tests for some of the stranger edge cases here.

If full round-trip tests (eg. CSV -> objects) are too complex, then split some of the crucial code into separate methods on the object, and test them directly. eg. the media summary file code can provide a row and get back an object, which is easier to write a unit test for.

File: peachjam/resources.py (hunk 2 above)
Comment (actlikewill) [reply peachjam/resources.py]: ✔️

File: peachjam/resources.py (hunk 1 above)
Comment (actlikewill) [reply peachjam/resources.py]: This is supposed to strip white space around and between the judges name to avoid the duplication issue we encountered. I have changed it to a list comprehension.

File: peachjam/resources.py [hunk 5]
```diff
@@ -146,31 +163,51 @@ def before_import_row(self, row, **kwargs):
                 defaults={"name": row["court_obj"]["name"]},
             )
 
+        source_files = row["source_url"].split("|")
```
[the remark this answers is not in the document]
Comment (actlikewill) [reply peachjam/resources.py]: changed.

File: peachjam/resources.py (hunk 3 above)
Comment (actlikewill) [reply peachjam/resources.py]: I have added  a break in the loop to terminate when it encounters a .docx file. Otherwise it will accept a pdf file if it doesn't find a docx.

File: peachjam/resources.py (hunk 4 above)
Comment (actlikewill) [reply peachjam/resources.py]: added.

**Review by longhotsummer — APPROVED** [review]
